### PR TITLE
gitoxide: 0.3.0 -> 0.4.0

### PR DESCRIPTION
**Pull request by syberant**
###### Motivation for this change
To update the package I'm maintaining.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/gitoxide/default.nix [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+  # Needed to get openssl-sys to use pkgconfig.
+  OPENSSL_NO_VENDOR = 1;
 
   meta = with lib; {
     description =
       "A command-line application for interacting with git repositories";
     homepage = "https://github.com/Byron/gitoxide";
-    # NOTE: the master branch is dual-licensed with APACHE but v0.3.0 is only MIT
-    license = licenses.mit;
+    license = with licenses; [ mit asl20 ];
```
Comment (danieldk) [inline pkgs/applications/version-management/gitoxide/default.nix:26]:
```suggestion
    license = with licenses; [ mit /* or */ asl20 ];
```

Which it the convention if either license can be used.

File: pkgs/applications/version-management/gitoxide/default.nix [hunk 2]
```diff
@@ -1,24 +1,29 @@
-{ lib, rustPlatform, fetchFromGitHub }:
+{ lib, rustPlatform, fetchFromGitHub, openssl, pkgconfig }:
```
Comment (danieldk) [inline pkgs/applications/version-management/gitoxide/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, pkg-config, openssl }:
```

`pkgconfig` is an alias for `pkg-config`. Also, we generally order `nativeBuildInputs` before `buildInputs`.

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks! Only two small suggested changes.

**Comment by syberant** [comment]
Applied and squashed, thanks for the feedback!

**Review by danieldk — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 97893` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitoxide</li>
  </ul>
</details>

**Pull request merged**